src/amuse/units/generic_unit_converter.py

```diff
@@ -113,15 +113,15 @@ def matrixrank(self, A, tol=1e-8):
         return numpy.sum(numpy.where( s>tol, 1, 0 ) )
 
     def determine_new_base(self):
-        matrix = numpy.mat(numpy.zeros((self.system_rank,self.system_rank)))
+        matrix = numpy.asmatrix(numpy.zeros((self.system_rank,self.system_rank)))
```

Comment:
I think pytest is warning that matrix is deprecated? should be ordinary array and then the matrix operations should be replaced ( * -> .dot( and **-1 -> numpy.linalg.inv())

Reply:
ya. sounds good.

Reply:
matrix is not yet deprecated but "pending deprecation" and not recommended. 
asmatrix and mat are the same, I found asmatrix easier to find when it needs to be replaced.

Reply:
this is not immediately necessary so I'll leave it for now